database/test/db-admin-persona.test.js

```diff
@@ -1,45 +1,53 @@
 const { describe, it } = require('mocha')
 const assert = require('assert')
 const postPersonas = require('../layer/admin-persona.js')
+const { usherDb } = require('../layer/knex')
 
-describe('Admin persona view', function () {
-  describe('Test INSERT personas', function () {
-    it('Should insert persona without an exception', async function () {
+describe('Admin persona view', () => {
+  describe('Test INSERT personas', () => {
+    it('Should insert persona without an exception', async () => {
       const insertResult = await postPersonas.insertPersona('test-tenant1', 'http://idp.dmgt.com.mock.localhost:3002/', 'test-dmgt-oocto-1@dmgtoocto.com', '')
       assert.strictEqual(insertResult, 'Insert successful')
       await postPersonas.deletePersona('test-tenant1', 'http://idp.dmgt.com.mock.localhost:3002/', 'test-dmgt-oocto-1@dmgtoocto.com', '')
     })
-    it('Should fail to insert for a nonexistent tenant', async function () {
+    it('Should fail to insert for a nonexistent tenant', async () => {
       const insertResult = await postPersonas.insertPersona('test-tenant1 Non-existent', 'http://idp.dmgt.com.mock.localhost:3002/', 'test-dmgt-oocto-3@dmgtoocto.com', '')
       assert.strictEqual(insertResult, 'Insert failed: Tenant does not exist matching tenantname test-tenant1 Non-existent iss_claim http://idp.dmgt.com.mock.localhost:3002/')
     })
-    it('Should fail to insert duplicate tenant/persona combination - check tenantname', async function () {
+    it('Should fail to insert duplicate tenant/persona combination - check tenantname', async () => {
       await postPersonas.insertPersona('test-tenant1', 'http://idp.dmgt.com.mock.localhost:3002/', 'test-dmgt-oocto-x@dmgtoocto.com', '')
       const result = await postPersonas.insertPersona('test-tenant1', 'http://idp.dmgt.com.mock.localhost:3002/', 'test-dmgt-oocto-x@dmgtoocto.com', '')
       assert.strictEqual(result, 'Insert failed: A persona (sub_claim = test-dmgt-oocto-x@dmgtoocto.com; user_context = ) already exists on tenantname test-tenant1 iss_claim http://idp.dmgt.com.mock.localhost:3002/')
       await postPersonas.deletePersona('test-tenant1', 'http://idp.dmgt.com.mock.localhost:3002/', 'test-dmgt-oocto-x@dmgtoocto.com', '')
     })
+    it('Should insert persona by tenant key without an exception', async () => {
+      const subClaim = 'test-user@the-usher.com'
+      const [tenant] = await usherDb('tenants').select('*').limit(1)
```

Comment:
Should we put this responsibility on the endpoint?

Reply:
Can you elaborate? This is an integration test to check the functionality of `insertPersonaByTenantKey` function.

Reply:
hey @k1ch , I was actually asking for this endpoint to check valid tenant key first and return a 400 as bad request. Looks like we all ready have a foreign key constraint for the same on personas table.